core/camera/src/main/java/com/google/jetpackcamera/core/camera/CoroutineCameraProvider.kt

```diff
@@ -36,7 +37,7 @@ import kotlinx.coroutines.coroutineScope
 suspend fun <R> ProcessCameraProvider.runWith(
     cameraSelector: CameraSelector,
     useCases: UseCaseGroup,
-    block: suspend (Camera) -> R
+    block: suspend CoroutineScope.(Camera) -> R
```

Comment:
This code, originating from the initial commit of pull request #251 by @temcguir , aims to fix a [bug](https://github.com/google/jetpack-camera-app/pull/250#discussion_r1696138031) where coroutines within the camera scope weren't canceled when switching cameras.

The intention, as I understand it, is to have the `block` function call occur within the coroutine scope established by `runWith`, specifically inside `= coroutineScope { ... }`. This would ensure that any launch calls within the `block` are also cancelled when the `runWith` block is cancelled. The actual cancellation is triggered by the below `collectLatest` (refer to the [doc](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/collect-latest.html) for details) when new settings, such as camera switching, are detected, and this cancellation propagates to `runWith` and `block`.

```
currentSettings.
    }.distinctUntilChanged()
    .collectLatest {
         // ...
    }
```